pkg/controller/direct/dataflow/dataflowflextemplatejob_controller.go

```diff
@@ -330,30 +333,54 @@ func (a *dataflowFlexTemplateJobAdapter) Create(ctx context.Context, createOp *d
 	}
 
 	job := response.GetJob()
-	jobID := job.GetId()
-	// TODO: Use jobCreateTime for mutation checking?
-	// jobCreateTime := job.CreateTime()
 
-	ready := false
-	for !ready {
-		time.Sleep(2 * time.Second)
-		latest, err := a.getJob(ctx, jobID)
-		if err != nil {
-			// TODO: not right!
-			return fmt.Errorf("getting state of job")
+	if err := a.updateStatus(ctx, createOp, job); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (a *dataflowFlexTemplateJobAdapter) updateStatus(ctx context.Context, op directbase.Operation, job *pb.Job) error {
+	status := &krm.DataflowFlexTemplateJobStatus{
+		JobID:        job.GetId(),
+		CurrentState: direct.PtrTo(job.CurrentState.String()),
+		Type:         direct.PtrTo(job.Type.String()),
+	}
+
+	var readyCondition *v1alpha1.Condition
+
+	switch job.CurrentState {
+	case pb.JobState_JOB_STATE_RUNNING:
```

Comment:
I think pb.JobState_JOB_STATE_UPDATED also applies to this case,  right?

Reply:
Agreed, added a bunch more terminal but not ready states.

Also made the requeuing explicit - it was just too magic.  This means we can set the ready condition however we best see fit.  I think the current ready logic is right - on a terminal condition we stop reconciling but ready=false, but IDK!